MENTAL-MODEL.md

```diff
@@ -0,0 +1,376 @@
+
+### 7. Sign and finalize the Payjoin Proposal PSBT
+
+Most software can handle adding the last signatures to a PSBT without issue.
+
+```rs
+let psbt = bitcoind
+    .wallet_process_psbt(&serialize_psbt(&psbt), None, None, None)
+    .with_context(|| "Failed to process PSBT")?
+    .psbt;
+let tx = bitcoind
+    .finalize_psbt(&psbt, Some(true))
+    .with_context(|| "Failed to finalize PSBT")?
+    .hex
+    .ok_or_else(|| anyhow!("Incomplete PSBT"))?;
+```
+
+### 8. Broadcast the Payjoin Transaction
+
+In order to preserve privacy between the transaction and the IP address from which it originates, transaction broadcasting should be done using Tor, a VPN, or proxy.
+
+```rs
+let txid =
+    bitcoind.send_raw_transaction(&tx).with_context(|| "Failed to send raw transaction")?;
+log::info!("Transaction sent: {}", txid);
+```
+
+📤 Sending payjoin is just that simple.
+
+## Receive a Payjoin
+
+The `receiver` feature provides all of the check methods, PSBT data manipulation, coin selection, and transport structures to receive payjoin and handle errors in a privacy preserving way.
+
+Receiving payjoin entails listening to a secure http endpoint for inbound requests. The endpoint is displayed in the `pj` parameter of a bip21 request URI.
+
+The reference implementation uses `rouille` sync http server and Bitcoin Core RPC.
+
+```rs
+fn receive_payjoin(
+    bitcoind: bitcoincore_rpc::Client,
+    amount_arg: &str,
+    endpoint_arg: &str,
+) -> Result<()>
+```
+
+### 1. Generate a pj_uri BIP21 using `payjoin::Uri::from_str`
+
+A BIP 21 URI supporting payjoin contains at minimum a bitcoin address and a secure `pj` endpoint.
+
+```rs
+let pj_receiver_address = bitcoind.get_new_address(None, None)?;
+let amount = Amount::from_sat(amount_arg.parse()?);
+let pj_uri_string = format!(
+    "{}?amount={}&pj={}",
+    pj_receiver_address.to_qr_uri(),
+    amount.to_btc(),
+    endpoint_arg
+);
+let pj_uri = Uri::from_str(&pj_uri_string)
+    .map_err(|e| anyhow!("Constructed a bad URI string from args: {}", e))?;
+let _pj_uri = pj_uri
+    .check_pj_supported()
+    .map_err(|e| anyhow!("Constructed URI does not support payjoin: {}", e))?;
+```
+
+### 2. Listen for a sender's request on the `pj` endpoint
+
+Start a server to respond to payjoin protocol POST messages.
+
+```rs
+rouille::start_server("0.0.0.0:3000", move |req| handle_web_request(&req, &bitcoind));
+// ...
+fn handle_web_request(req: &Request, bitcoind: &bitcoincore_rpc::Client) -> Response {
+    handle_payjoin_request(req, bitcoind)
+        .map_err(|e| match e {
+            ReceiveError::RequestError(e) => {
+                log::error!("Error handling request: {}", e);
+                Response::text(e.to_string()).with_status_code(400)
+            }
+            e => {
+                log::error!("Error handling request: {}", e);
+                Response::text(e.to_string()).with_status_code(500)
+            }
+        })
```

Comment:
It would be a better experience to respond to HTTP GET requests to this endpoint from browsers with a payjoin QuickStart HTML page instead of 404 or 500 errors.